src/libs/OptionsListUtils.ts

```diff
@@ -2279,12 +2294,14 @@ function filterOptions(options: Options, searchInputValue: string): Options {
                 if (item.alternateText) {
                     values.push(item.alternateText);
                 }
+                values = values.concat(getParticipantsLoginsArray(item));
             } else if (!!item.isChatRoom || !!item.isPolicyExpenseChat) {
                 if (item.subtitle) {
                     values.push(item.subtitle);
                 }
+            } else {
+                values = values.concat(getParticipantsLoginsArray(item));
```

Comment:
Why do we move this to the last else block, with that for the policy expense chat it does not consider the participants but only subtitle for the filter values. Due to that policy expense chats are missing in the Recent section while filtering.

https://github.com/Expensify/App/assets/14358475/177aad7a-4279-472f-b13c-0f2ae94eff6c

Reply:
Recently this PR caused a big performance regression: #38887 and it was reverted in #40019. Since the filtering was merged somewhere close before reverting, I adjusted function to match the correct behavior. Seems like it works the same way on production

Reply:
Ah, Ok. On prod I am not seeing those policy expense chat for the global search but it is shown for the request/send money participants filtration. I think that reverted PR tried manipulating the `getSearchText` but here for request money participants search I don't think we are making a call to `getSearchText`.

https://github.com/Expensify/App/assets/14358475/71ab3e8d-dba6-4894-affb-c2657769dab0

Reply:
That reverted PR is in production, doubt why it still shows the policy expense chats in the search filter for request/send money flow.

<img width="1440" alt="Screenshot 2024-04-22 at 01 00 19" src="https://github.com/Expensify/App/assets/14358475/5b45af03-1850-407e-a9bb-ce131666b68f">

Reply:
I re-reviewed the `isSearchStringMatch` function, responsible for searching a match in a string generated by `getSearchText` and found that for non-chatrooms, it also looks for matches in a participants list. Updated the `filterOptions` to match this behavior so now it should also display e.g. workspaces when you search by user name. Thanks for spotting this! 👍